### PR TITLE
modify canvas_page.py so it also works on Linux

### PR DESCRIPTION
**Pull request by qq1990**
Use "pack geometry" and scrollable instead of "grid()" structure

### CODE REVIEW

File: virgo/ui/pages/canvas_page.py [hunk 1]
```diff
@@ -42,89 +42,119 @@ def Page(self: App):
...
     self.widgetMenu.grid_columnconfigure(0, weight=1)
-
-    widgets = scrollable_frame.ScrollableFrame(self.widgetMenu)
-    widgetsFrame = widgets.get()
     
-    #List of different widgets to add to the canvas
-    ttk.Label(widgetsFrame, text="Data:", style="H3.TLabel").grid()
-    ttk.Button(widgetsFrame, text="Add a Data Source", command=lambda x=base_nodes.DataSourceNode: self.add_source_node_to_canvas(x)).grid()
+    # Create a canvas within the widgetMenu to hold the scrollable content
+    canvas = tk.Canvas(self.widgetMenu)
```
Comment (GeorgeHuber) [inline virgo/ui/pages/canvas_page.py]: Could we name this something more descriptive and refactor the scrolling code into the class here, similar to how it was setup before? Just so it's not redundant when we add more tabs [Scrollable frame](https://github.com/GeorgeHuber/virgo/blob/main/virgo/ui/components/scrollable_frame.py)

File: virgo/ui/pages/canvas_page.py [hunk 2]
```diff
@@ -42,89 +42,119 @@ def Page(self: App):
...
+    self.widgetMenu.pack(fill="both", expand=True, padx=10, pady=10)
 
     # Configurations Code
     self.configMenu.grid_rowconfigure(0, weight=1)
     self.configMenu.grid_columnconfigure(0, weight=1)
-    config = scrollable_frame.ScrollableFrame(self.configMenu)
-    configFrame = config.get()
-
     
+    # Create a canvas within the configMenu to hold the scrollable content
```
Comment (GeorgeHuber) [inline virgo/ui/pages/canvas_page.py]: same here, could we move this into a component instead?

File: virgo/ui/pages/canvas_page.py [hunk 3]
```diff
@@ -12,25 +12,25 @@
 if TYPE_CHECKING:
     from virgo.app import App
 
-#TODO: turn this into a class and refactor
```
Comment (GeorgeHuber) [inline virgo/ui/pages/canvas_page.py]: Lets keep all the #TODO comments for now, I'll put them in a list and remove later.

File: virgo/ui/pages/canvas_page.py [hunk 4]
```diff
@@ -42,89 +42,119 @@ def Page(self: App):
...
     
-    widgets.grid(column=0, row=0, sticky="nsew")
-    config.grid(column=0, row=0, sticky="nsew")
+    # Bind the scroll functionality to the mouse wheel
+    canvas_config.bind_all("<MouseWheel>", on_mousewheel)
+
+    # Configure the configMenu frame to expand
+    self.configMenu.pack(fill="both", expand=True, padx=10, pady=10)
 
     def tab_change_handler(event):
```
Comment (GeorgeHuber) [inline virgo/ui/pages/canvas_page.py]: there seems to be an issue on mac when changing tabs that the content in the configurations tabs will not load until the mouse is moved over it. Does this behavior reproduce for you?

**Comment by GeorgeHuber** [comment]
Hey Quan, looking good! Thanks for working on this and cleaning up some of the hackathon style code in the process. Just a few comments, let me know your thoughts!

File: virgo/ui/pages/canvas_page.py (hunk 4 above)
Comment (qq1990) [reply virgo/ui/pages/canvas_page.py]: This is weird. I haven't encountered that issue on my Mac. The content in the configuration tabs loads immediately when switching between tabs.

**Pull request closed without merging**